src/sentry/static/sentry/app/components/events/interfaces/spans/filter.tsx

```diff
@@ -0,0 +1,342 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {t, tn} from 'app/locale';
+import space from 'app/styles/space';
+import {IconFilter} from 'app/icons';
+import DropdownControl from 'app/components/dropdownControl';
+import DropdownButton from 'app/components/dropdownButton';
+import CheckboxFancy from 'app/components/checkboxFancy/checkboxFancy';
+import {pickSpanBarColour} from 'app/components/events/interfaces/spans/utils';
+
+import {ParsedTraceType} from './types';
+
+type DropdownButtonProps = React.ComponentProps<typeof DropdownButton>;
+
+type NoFilter = {
+  type: 'no_filter';
+};
+
+type ActiveFilter = {
+  type: 'active_filter';
+  operationNames: Set<string>;
+};
+
+export const noFilter: NoFilter = {
+  type: 'no_filter',
+};
+
+export type ActiveOperationFilter = NoFilter | ActiveFilter;
+
+type Props = {
+  parsedTrace: ParsedTraceType;
+  operationNameFilters: ActiveOperationFilter;
+  toggleOperationNameFilter: (operationName: string) => void;
```

Comment:
it's just a suggestion but wouldn't be better if you add` type OperationName = string` and replace  `operationNames: Set<string>;` with  `operationNames: Set<OperationName>; `